### DISCUSSION TITLE
Redirect exceptions for specific paths

### DISCUSSION THREAD

**Discussion by mxstbr** [vercel/next.js#25727] [Help]
I have a `pages` folder like this:

```
pages
    create.tsx
    settings.tsx
    [slug]
        overview.tsx
```

I want to redirect `/:slug` to `/:slug/overview`, so I added this redirect:

```JS
// next.config.js

module.exports = {
  async redirects() {
    return [
      {
        source: '/:slug',
        destination: '/:slug/overview'
        permanent: true,
      },
    ]
  }
}
```

But now, of course, `/create` and `/settings` redirect to `/create/overview` and `/settings/overview` — which I obviously don't want. **How can I except these two paths from the redirect?**

Things I've tried that didn't work:

- Adding redirects for `/create` and `/settings` that redirect to themselves earlier in the array
- Trying to hack the `source` with regex like this: `/:slug((?!settings|create)\\w*)`

**Comment by carsoncychen** [comment]
Running into the exact same issue. Any suggestion here?

**Comment by carsoncychen** [comment]
Able to resolve this by using regex:

`source: "/:slug(\\w{8}-\\w{4}-\\w{4}-\\w{4}-\\w{12})"`
